internal/manifests/mutate.go

```diff
@@ -82,3 +90,8 @@ func mutateStatefulSet(existing, desired *appsv1.StatefulSet) {
 	existing.Spec.Template = desired.Spec.Template
 	existing.Spec.VolumeClaimTemplates = desired.Spec.VolumeClaimTemplates
 }
+
+func mutateServiceMonitor(existing, desired *monitoringv1.ServiceMonitor) {
+	// ServiceMonitor selector is immutable so we set this value only if
+	// a new object is going to be created
+}
```

Comment:
Does this imply that we don't support mutation of service monitor resources? Or is this a hindsight?

Reply:
Yeah, it's kind of hindsight as of now.

Reply:
Why? Is this going to stay empty?

Reply:
Currently we don't want to provide any dynamic updation from users so we don't need to mutate. Hence it should be empty right?

Reply:
As per now, we don't really manipulate anything in the ServiceMonitors, you are right. They are tightly coupled to a LokiStack name and thus not mutable. I tend to err that this can stay empty for now, as we might have to handle TLSConfig changes introduced by #59

Reply:
So can we let this stay empty now? Once TLSConfig PR is merged, I will create another task in which we can work on the loki-operator CLI flag (`utilruntime.Must(monitoringv1.AddToScheme(scheme))`) optional and adding the TLSConfig changes.